### PR TITLE
expression: builtin function current_date() and curdate() should return DATE t…

### PR DESCRIPTION
**Pull request by hawkingrei**
Signed-off-by: Weizhen Wang <wangweizhen@pingcap.com>

### What problem does this PR solve?

builtin function ```current_date()``` and ```curdate()``` should return DATE type.

Issue Number:  #24929 

Problem Summary:

### What is changed and how it works?

```current_date()``` and ```curdate()``` return DATE type.

### Check List 

Tests 

- Unit test

### Release note

- No Release Note

### CODE REVIEW

File: expression/typeinfer_test.go [hunk 1]
```diff
@@ -1698,8 +1698,9 @@ func (s *testInferTypeSuite) createTestCase4TimeFuncs() []typeInferTestCase {
 		{"utc_time(6) ", mysql.TypeDuration, charset.CharsetBin, mysql.BinaryFlag | mysql.NotNullFlag, 15, 6},
 		{"utc_time(7) ", mysql.TypeDuration, charset.CharsetBin, mysql.BinaryFlag, 15, 6},
 
-		{"utc_date()  ", mysql.TypeDatetime, charset.CharsetBin, mysql.BinaryFlag | mysql.NotNullFlag, 10, 0},
-		{"curdate()", mysql.TypeDatetime, charset.CharsetBin, mysql.BinaryFlag | mysql.NotNullFlag, 10, 0},
+		{"utc_date() ", mysql.TypeDatetime, charset.CharsetBin, mysql.BinaryFlag | mysql.NotNullFlag, 10, 0},
```
Comment (ichn-hu) [inline expression/typeinfer_test.go:1701]: why is this line modified?

Comment (hawkingrei) [reply expression/typeinfer_test.go:1701]: remove extra space。

**Comment by bb7133** [comment]
Thanks!

Please update the title according to: https://github.com/pingcap/community/blob/master/contributors/commit-message-pr-style.md#commit-message-and-pull-request-style

File: expression/typeinfer_test.go (hunk 1 above)
Comment (wjhuang2016) [reply expression/typeinfer_test.go:1701]: refine the padding space?

**Review by hawkingrei — COMMENTED** [review]
> Thanks!
> 
> Please update the title according to: https://github.com/pingcap/community/blob/master/contributors/commit-message-pr-style.md#commit-message-and-pull-request-style

```expression:``` instead of ```fix(expression):```

**Review by wjhuang2016 — APPROVED** [review]

File: expression/typeinfer_test.go (hunk 1 above)
Comment (hawkingrei) [reply expression/typeinfer_test.go]: I have formated this code, make it padding.

**Comment by hawkingrei** [comment]
> Thanks!
> 
> Please update the title according to: https://github.com/pingcap/community/blob/master/contributors/commit-message-pr-style.md#commit-message-and-pull-request-style

```expression: ```instead of ```fix(expression):``` @bb7133

**Review by ichn-hu — APPROVED** [review]
